### PR TITLE
[CELEBORN-1641] Pack the output spark client jar with the major.minor version

### PR DESCRIPTION
**Pull request by zaynt4606**
### What changes were proposed in this pull request?
as title

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
sbt:
image
image
maven:
image

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -67,7 +67,7 @@
     <maven.version>3.9.9</maven.version>
 
     <flink.version>1.14.6</flink.version>
-    <spark.version>3.3.4</spark.version>
+    <spark.version>3.4.4</spark.version>
```
Comment (zaynt4606) [inline pom.xml:70]: The IDEA Maven plugin prioritizes the outer properties in the artifactId. 
Therefore, we need to update the version in the pom file when using the IDEA Maven plugin. 
This update does not impact the compilation process.

**Comment by zaynt4606** [comment]
cc @SteNicholas @turboFei @FMX @RexXiong

File: build/make-distribution.sh [hunk 2]
```diff
@@ -185,11 +185,10 @@ function build_spark_client {
       | grep -v "WARNING" \
       | tail -n 1)
   SPARK_MAJOR_VERSION=${SPARK_VERSION%%.*}
-
```
Comment (turboFei) [inline build/make-distribution.sh:188]: unnecessary change

File: pom.xml (hunk 1 above)
Comment (turboFei) [reply pom.xml:70]: The spark version format should be majorVersion.minorVersion.bugfixVersion

How about using `spark.binary.version` property with majorVersion.minorVersion?

File: build/make-distribution.sh (hunk 2 above)
Comment (zaynt4606) [reply build/make-distribution.sh:188]: Thanks!

File: pom.xml (hunk 1 above)
Comment (zaynt4606) [reply pom.xml:70]: I am fine with it~

**Review by pan3793 — CHANGES_REQUESTED** [review]
why

**Comment by zaynt4606** [comment]
> why

Display the version of spark compiled by the client as in client-flink

**Comment by pan3793** [comment]
then deploy dozens of spark client jars into the final release binary tgz? which virtually are the same one.

I don't know how frequently the Flink API changes, but the Spark side API we use is pretty stable. To record the build env info, you can follow CELEBORN-1658's approach.

**Comment by zaynt4606** [comment]
> then deploy dozens of spark client jars into the final release binary tgz? which virtually are the same one.
> 
> I don't know how frequently the Flink API changes, but the Spark side API we use is pretty stable. To record the build env info, you can follow [CELEBORN-1658](https://issues.apache.org/jira/browse/CELEBORN-1658)'s approach.

There will be still one spark client jar and just want to show the spark version info in jar name.

Our own version of spark will need this information and I think it would be nice for the community to see this information in the client jar. But in fact they are all the same one.

**Comment by pan3793** [comment]
From a user perspective, I would think that `abc-spark-3.4.3-xyz.jar` is only applicable to Spark 3.4.3.

**Comment by zaynt4606** [comment]
> From a user perspective, I would think that `abc-spark-3.4.3-xyz.jar` is only applicable to Spark 3.4.3.

Yes, this pull request needs to explain in the document. And there is a trade-off between providing this information and ensuring that it remains easy for users to understand. I'd like to know what the community thinks about this.

**Comment by pan3793** [comment]
> And there is a trade-off between providing this information and ensuring that it remains easy for users to understand.

the jar name should reflect the compatible runtime spark version instead of the exact spark version at building time.

In short, I'm -1 on this change, and already provided an alternative solution

> To record the build env info, you can follow [CELEBORN-1658](https://issues.apache.org/jira/browse/CELEBORN-1658)'s approach.

**Comment by RexXiong** [comment]
> > And there is a trade-off between providing this information and ensuring that it remains easy for users to understand.
> 
> the jar name should reflect the compatible runtime spark version instead of the exact spark version at building time.
> 
> In short, I'm -1 on this change, and already provided an alternative solution
> 
> > To record the build env info, you can follow [CELEBORN-1658](https://issues.apache.org/jira/browse/CELEBORN-1658)'s approach.

If that's the case, can we remove all the spark-3.x profiles in the pom and only keep one? These profiles are very confusing.

**Comment by pan3793** [comment]
> can we remove all the spark-3.x profiles in the pom and only keep one?

I don't opposite that as long as you build an integration test pipeline to cover all supported Spark versions before removing. For example, the released Kyuubi Spark/Flink engine jar does not contain Spark/Flink engine version, and it guarantees that the packaged jar can run across all supported Spark/Flink versions, by https://github.com/apache/kyuubi/blob/b4838b40e6a9074697918ccecd3dfe71cc52442d/.github/workflows/master.yml#L63-L82

**Comment by RexXiong** [comment]
> > can we remove all the spark-3.x profiles in the pom and only keep one?
> 
> I don't opposite that as long as you build an integration test pipeline to cover all supported Spark versions before removing. For example, the released Kyuubi Spark/Flink engine jar does not contain Spark/Flink engine version, and it guarantees that the packaged jar can run across all supported Spark/Flink versions, by https://github.com/apache/kyuubi/blob/b4838b40e6a9074697918ccecd3dfe71cc52442d/.github/workflows/master.yml#L63-L82

Thanks, seems we need use the version to be released to verify all supported spark versions at least during the testing process.
